### DISCUSSION TITLE
Custom route badges

### DISCUSSION THREAD

**Discussion by arthurvanl** [scalar/scalar#3708] [Ideas]
Proposing custom badges for routes. These badges will be next to the method badge:
![image](https://github.com/user-attachments/assets/51ab961c-61a3-4100-9c62-0a3951eeafa2)

This makes it easy to show when new routes are added into your openapi docs

![image](https://github.com/user-attachments/assets/111ca3b1-f66c-4450-b0ba-abef5a59fa0d)

Example of how a path object would look like:
```json
"/museum-hours": {
    "get": {
      "summary": "Get museum hours",
      "description": "Get upcoming museum operating hours",
      "operationId": "getMuseumHours",
      "x-badges": [
        {
          "name": "Beta",
          "position": "before",
          "color": "purple"
        }
      ]
   }
}
```

as you can see, I added the color `purple`, but maybe we should make it so this is just a hexcolor.

**Comment by hanspagel** [comment]
I kinda like the idea. I think the position shouldn’t be configurable though.

**Reply by arthurvanl** [reply]
Any updates on this?

**Reply by WDaan** [reply]
I've took a stab at it in https://github.com/scalar/scalar/pull/4463
